BLL/Services/DirectoryService.cs

```diff
@@ -88,19 +101,41 @@ public CityDTO GetCityById(int cityId)
             return _mapper.Map<City, CityDTO>(_db.Cities.Get(cityId));
         }
 
+        public string GetLocationById(int cityId)
+        {
+            var city = _db.Cities.Get(cityId);
```

Comment:
no need to make 3 calls to the DB, you can do it with only one:

var city = _db.Cities.FindWithSpecificationPattern(new CitySpecification()).FirstOrDefault(x => x.Id == cityId);
return $"{city.State.Country.CountryName}, {city.State.StateName}, {city.CityName}";